### PR TITLE
[#9293] Stateless/Agent/Beat Boiler Plate for `pause-orchestration` annotation

### PR DESCRIPTION
**Pull request by dtuck9**
## Summary
                                                                                                                                                                                                                                                                                                                            
  This PR lays the groundwork for supporting the `eck.k8s.elastic.co/pause-orchestration` annotation on Agent and Beat resources (issue #9293). It is the stateless/Agent/Beat counterpart to the Elasticsearch implementation already in place. This is almost entirely boiler plate as a result of adding `Conditions` to the `DeploymentStatus`, the `AgentStatus`, and the `BeatStatus` and running `make generate`. 

The boiler plate code distracted from the primary logic changes in #9398, so I opted to break it apart into more digestible chunks.                                                                                                                                                                                                                                        
                                                                                                                                                                                                                                                                                                                            
  ### What changed                                                                                                                                                                                                                                                                                                          
                                                                                                                                                                                                                                                                                                                            
  **New `pause_orchestration.go` in `pkg/controller/common`**                                                                                                                                                                                                                                                               
   
  `unmanaged.go` into a dedicated `pause_orchestration.go` file in the same package. This separates pause-orchestration logic from the deprecated `eck.k8s.elastic.co/managed` annotation and gives it a natural home alongside the reconciliation helpers that will follow in #9398. Tests move with it into `pause_orchestration_test.go`. This allows `unmanaged.go` to be removed once the `managed` annotation is removed in a few releases.                                                                                                                                                                                                                                                      
                                                                                                                                                                                                                                                                                                                            
  **`Conditions` field added to `AgentStatus`, `BeatStatus`, and `DeploymentStatus`**                                                                                                                                                                                                                                                       
                                                                                                                                                                                                                                                                                                                            
  `AgentStatus`, `BeatStatus`, and the shared `DeploymentStatus` (used by APMServer, Kibana, EnterpriseSearch, ElasticMapsServer, and PackageRegistry) now carry a `Conditions commonv1.Conditions` field. The corresponding CRD manifests and `zz_generated.deepcopy.go` files are updated accordingly via `make generate`.                                                                                                                                                                                                                                                                 
                                                                                                                                                                                                                                                                                                                            
  ### What this does not yet include                                                                                                                                                                                                                                                                                        
                                                                                                                                                                                                                                                                                                                            
  The actual pause-check logic in the Agent and Beat reconcile loops — i.e. calling `IsOrchestrationPaused` and setting the `OrchestrationPaused` condition — is not wired up in this PR. This PR is intentionally scoped to the API and infrastructure changes needed before that work can land cleanly with a tighter scope.                                                                                                                                                                                                                                                                         
                                                                                                                                                                                                                                                                                                                            
  ## Testing                                                                                                                                                                                                                                                                                                                
                                                                                                                                                                                                                                                                                                                            
  - Existing unit tests for `IsOrchestrationPaused` and `IsUnmanaged` are preserved and updated to reflect the file move.                                          
  - New unit tests in `pause_orchestration_test.go` cover the annotation check directly, including edge cases (empty string, non-`"true"` values, repeated toggling).                                                                                                                                                                                                                                            
                                                                                                                                                                                                                                                                                                                            
  ## Related                                                                                                                                                                                                                                                                                                                
                                                                                                                                                                                                                                                                                                                            
  - Issue: #9293                                                                                                                                                                                                                                                                                                            
  - Elasticsearch pause-orchestration implementation: already merged
  - Stateless controller changes (Kibana, APMServer, etc.): https://github.com/elastic/cloud-on-k8s/pull/9398

### CODE REVIEW

**Comment by prodsecmachine** [comment]
### :white_check_mark: **Snyk checks have passed. No issues have been found so far.**

| Status | Scan Engine |  Critical |  High |  Medium |  Low | Total (0) |
|---|---|---|---|---|---|---|
| :white_check_mark: | Open Source Security| 0 | 0 | 0 | 0 | [ 0 issues](https://app.snyk.io/org/elastic-cloud-on-k8s/pr-checks/2300f46a-7979-415b-a623-1e37024e3b7d?refs=b9bed527ea763264ed87e815d22d94a2d45cc1de&source=prChecksComment) |
| :white_check_mark: | Licenses| 0 | 0 | 0 | 0 | [ 0 issues](https://app.snyk.io/org/elastic-cloud-on-k8s/pr-checks/2300f46a-7979-415b-a623-1e37024e3b7d/license?refs=b9bed527ea763264ed87e815d22d94a2d45cc1de&source=prChecksComment) |

:computer: Catch issues earlier using the plugins for [VS Code](https://marketplace.visualstudio.com/items?itemName=snyk-security.snyk-vulnerability-scanner), [JetBrains IDEs](https://plugins.jetbrains.com/plugin/10972-snyk-security), [Visual Studio](https://marketplace.visualstudio.com/items?itemName=snyk-security.snyk-vulnerability-scanner-vs-2022), and [Eclipse](https://marketplace.eclipse.org/content/snyk-security#external-install-button).

File: test/e2e/test/enterprisesearch/checks_k8s.go [hunk 1]
```diff
@@ -133,11 +134,12 @@ func CheckStatus(b Builder, k *test.K8sClient) test.Step {
 					AvailableNodes: b.EnterpriseSearch.Spec.Count,
 					Version:        b.EnterpriseSearch.Spec.Version,
 					Health:         "green",
+					Conditions:     commonv1.Conditions{},
 				},
 				ExternalService: b.EnterpriseSearch.Name + "-ent-http",
 				Association:     commonv1.AssociationEstablished,
 			}
-			if ent.Status != expected {
+			if reflect.DeepEqual(ent.Status, expected) {
```
Comment (naemono) [inline test/e2e/test/enterprisesearch/checks_k8s.go]:
```suggestion
			if !reflect.DeepEqual(ent.Status, expected) {
```

I think all of these assertions here are backwards.

**Review by naemono — CHANGES_REQUESTED** [review]

File: test/e2e/test/enterprisesearch/checks_k8s.go (hunk 1 above)
Comment (dtuck9) [reply test/e2e/test/enterprisesearch/checks_k8s.go]: Wow good catch. But more concerningly, they passed the smoke test 🤔

Comment (naemono) [reply test/e2e/test/enterprisesearch/checks_k8s.go]: > Wow good catch. But more concerningly, they passed the smoke test 🤔

The smoke test only runs a limited set of e2e tests. You have to manually run the full set, or a targeted set of e2e tests.

**Comment by dtuck9** [comment]
buildkite test this -f p=gke,t=.*,E2E_TAGS=kb

[[build](https://buildkite.com/elastic/cloud-on-k8s-operator/builds/14641)]

File: pkg/controller/common/pause_orchestration_test.go [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+// you may not use this file except in compliance with the Elastic License 2.0.
+
+package common
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	corev1 "k8s.io/api/core/v1"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
```
Comment (pkoutsovasilis) [inline pkg/controller/common/pause_orchestration_test.go]: nit, let's be consistent with other import aliases
```suggestion
	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
```

File: pkg/controller/common/unmanaged_test.go
Comment (pebrc) [inline pkg/controller/common/unmanaged_test.go]: The shared testcase struct lives here, but TestIsOrchestrationPaused over in pause_orchestration_test.go now also depends on it. Per the PR description, unmanaged.go (and presumably unmanaged_test.go) is intended to be deleted in a subsequent PR — at that point the new test file would lose its testcase definition. Suggest moving the struct into pause_orchestration_test.go now so the deletion is a clean drop. Also worth refreshing the // Expected (un)managed state. comment since it's now shared.

File: pkg/apis/agent/v1alpha1/agent_types.go [hunk 3]
```diff
@@ -194,6 +194,8 @@ type AgentStatus struct {
 	// If the generation observed in status diverges from the generation in metadata, the Elastic
 	// Agent controller has not yet processed the changes contained in the Elastic Agent specification.
 	ObservedGeneration int64 `json:"observedGeneration,omitempty"`
+	// Conditions holds the current service state of the agent resource.
+	Conditions commonv1.Conditions `json:"conditions,omitempty"`
```
Comment (pebrc) [inline pkg/apis/agent/v1alpha1/agent_types.go]: Two small style nits relative to the existing ES Conditions field at pkg/apis/elasticsearch/v1/status.go:68:
   - Missing the // +optional kubebuilder marker that ES uses.
   - The json tag here is "conditions,omitempty" while ES uses "conditions" (no omitempty). Both work, but it means an empty Conditions will serialize differently across kinds. Probably because conditions already existed in the ES case? No big deal imo just wanted to make sure this was a conscious choice.
  
 (Same applies to BeatStatus and DeploymentStatus.)

File: test/e2e/test/apmserver/checks_k8s.go [hunk 4]
```diff
@@ -142,7 +143,7 @@ func CheckStatus(b Builder, k *test.K8sClient) test.Step {
 					Health:         "green",
 				},
 			}
-			if as.Status != expected {
+			if !reflect.DeepEqual(as.Status, expected) {
```
Comment (pebrc) [inline test/e2e/test/apmserver/checks_k8s.go]: Nit: test/e2e/test/beat/steps.go already uses `cmp.Equal` + `cmp.Diff` for nicer failure messages (cmp.Diff(actual, expected) produces a unified diff instead of two %+v dumps). Worth aligning the e2e check files touched here to the same style for consistency. Non-blocking.

**Review by pebrc — COMMENTED** [review]
LGTM

**Review by pebrc — APPROVED** [review]

File: pkg/apis/agent/v1alpha1/agent_types.go (hunk 3 above)
Comment (dtuck9) [reply pkg/apis/agent/v1alpha1/agent_types.go]: Good catch

Comment (dtuck9) [reply pkg/apis/agent/v1alpha1/agent_types.go]: @pebrc The `AutoscalingStatus` has `conditions,omitempty`. Would you like me to align that one as well?

Comment (dtuck9) [reply pkg/apis/agent/v1alpha1/agent_types.go]: I've left it out -- we can circle back to it separately if necessary.

**Comment by dtuck9** [comment]
buildkite test this -f p=gke,t=.*,E2E_TAGS=kb

[[build](https://buildkite.com/elastic/cloud-on-k8s-operator/builds/14658)]

**Review by naemono — APPROVED** [review]
I see nothing outstanding here. 👍
